src/yaml.rs

```diff
@@ -197,6 +210,42 @@ impl YamlLoader {
         parser.load(&mut loader, true)?;
         Ok(loader.docs)
     }
+
+    pub fn load_from_bytes(mut source: impl std::io::Read) -> Result<Vec<Yaml>, LoadError> {
+        let mut buffer = Vec::new();
+        source.read_to_end(&mut buffer)?;
+
+        // Decodes the input buffer using either UTF-8, UTF-16LE or UTF-16BE depending on the BOM codepoint.
+        // If the buffer doesn't start with a BOM codepoint, it will use a fallback encoding obtained by
+        // detect_utf16_endianness.
+        let (res, _) = encoding::types::decode(
+            &buffer,
+            encoding::DecoderTrap::Replace,
```

Comment:
@mkmik I'd like to understand why this decoder trap is sufficient in all cases. I was expecting it to be necessary to allow the user to choose.

Reply:
good point; probably strict is a better default and probably the user would want to be able to override this.

Reply:
(changed to Strict)

@glyn what's the sanest way to design an API that would let a user change the decoder error trapping behaviour? 

a. Should we make it a mandatory parameter?
b. An Option<>?
c. Two methods, one with defaults and one with some config structure that includes the encoding trap?
d. a "decoder" struct with a "decode" method and a builder pattern to set options:

```Decoder::read(file).encoding_trap(mytrap).decode()```

?

Reply:
I quite like (d) as it gives a clean API when you only want the default (strict) behaviour. I could live with (a) or (c). I don't like (b) as it requires an ugly `None` parameter for the default behaviour.